testing/compile/fork_join_cdp_deprecation/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+cmake_minimum_required(VERSION 3.15)
+
+# Remove this when we use the new CUDA_ARCHITECTURES properties with both
+# nvcc and nvc++.
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.18)
+  cmake_policy(SET CMP0104 OLD)
+endif()
+
+project(fork_join_cdp_deprecation CXX CUDA)
+
+message(STATUS "Thrust_DIR=${Thrust_DIR}")
+
+# Need C++14 for deprecation messages
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CUDA_STANDARD 14)
+
+find_package(Thrust)
+thrust_create_target(thrust)
+
+add_executable(main main.cu)
+target_link_libraries(main thrust)
+
+target_compile_definitions(main PUBLIC "EXEC_POLICY=${EXEC_POLICY}")
+
+if (IGNORE_DEPRECATION)
+  target_compile_definitions(main PUBLIC THRUST_IGNORE_DEPRECATED_FORK_JOIN_CDP)
+endif()
+
+# Need to enable RDC to enable deprecation warnings.
+if ("NVCXX" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
+  set_target_properties(main PROPERTIES
+    COMPILE_FLAGS "-gpu=rdc"
+  )
+else()
+  set_target_properties(main PROPERTIES
+    CUDA_SEPARABLE_COMPILATION ON
+  )
+endif()
+
+enable_testing()
+add_test(NAME main COMMAND $<TARGET_FILE:main>)
```

Comment:
If you rename the test to something like `test_main` you can use `COMMAND main` instead since you don't have a two targets with the same name.